io/tools/convert_clouds_meshes.cpp

```diff
@@ -0,0 +1,315 @@
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ \author Victor Lamoine
+ @b convert_point_clouds_meshes converts OBJ, PCD, PLY, STL, VTK files containing point clouds or meshes into every other format.
+ This tool allows to specify the file output type between ASCII, binary and binary compressed.
+ **/
+
+//TODO: Inform user about loss of color/alpha during conversion?
+// STL does not support color at all
+// OBJ does not support color in PCL (the format DOES support color)
+
+#include <vector>
+
+#include <pcl/console/parse.h>
+#include <pcl/io/auto_io.h>
+#include <pcl/io/obj_io.h>
+#include <pcl/io/vtk_lib_io.h>
+
+#include <boost/make_shared.hpp>
+
+#define ASCII 0
+#define BINARY 1
+#define BINARY_COMPRESSED 2
+
+/**
+ * Display help for this program
+ * @param argc[in]
+ * @param argv[in]
+ */
+void
+displayHelp (int argc,
+             char** argv)
+{
+  PCL_INFO ("\nUsage: %s [OPTION] SOURCE DEST\n", argv[0]);
+  PCL_INFO ("Convert SOURCE point cloud or mesh to DEST.\n\n");
+
+  PCL_INFO ("Available formats types for SOURCE and DEST:\n"
+           "\tOBJ (Wavefront)\n"
+           "\tPCD (Point Cloud Library)\n"
+           "\tPLY (Polygon File Format)\n"
+           "\tSTL (STereoLithography)\n"
+           "\tVTK (The Visualization Toolkit)\n\n");
+
+  PCL_INFO ("Available options:\n"
+           "\t-f, --format Specify DEST output type, available formats are ascii, binary and binary_compressed.\n"
+           "\t             When not specified, binary is used as default.\n"
+           "\t             OBJ only supports ascii format.\n"
+           "\t             binary_compressed is only supported by the PCD file format.\n");
+}
+
+bool
+saveMesh (pcl::PolygonMesh& input,
+          std::string output_file,
+          int output_type);
+
+/**
+ * Saves a cloud into the specified file and output type. The file format is automatically parsed.
+ * @param input[in] The cloud to be saved
+ * @param output_file[out] The output file to be written
+ * @param output_type[in] The output file type
+ * @return (true) on success, false otherwise.
+ */
+bool
+savePointCloud (pcl::PCLPointCloud2::Ptr input,
+                std::string output_file,
+                int output_type)
+{
+  if (boost::filesystem::path (output_file).extension () == ".obj")
+  {
+    //TODO: Support precision
+    //FIXME: Color is lost during conversion (OBJ supports color)
+    if (output_type == BINARY || output_type == BINARY_COMPRESSED)
+      PCL_WARN ("OBJ file format supports only ASCII.\n");
```

Comment:
This warning will also be printed in ```saveMesh()``` better to just remove it. This branch of your if statement can pretty much disappear and be moved to the else branch with ply and vtk.